### PR TITLE
executor: make `show stats_healthy` case-insensitive

### LINKED ISSUE #41544: STATS_HEALTHY being case sensitive
### 1. Minimal reproduce step (Required)

```sql
USE test;
CREATE TABLE t1(id INT PRIMARY KEY);
SHOW STATS_HEALTHY LIKE 'test%';
SHOW STATS_HEALTHY LIKE 'TEST%';
```

### 2. What did you expect to see? (Required)

```
sql> CREATE TABLE t1(id INT PRIMARY KEY);
Query OK, 0 rows affected (0.1738 sec)

sql> SHOW STATS_HEALTHY LIKE 'test%';
+---------+------------+----------------+---------+
| Db_name | Table_name | Partition_name | Healthy |
+---------+------------+----------------+---------+
| test    | t1         |                |     100 |
+---------+------------+----------------+---------+
1 row in set (0.0005 sec)

sql> SHOW STATS_HEALTHY LIKE 'TEST%';
+---------+------------+----------------+---------+
| Db_name | Table_name | Partition_name | Healthy |
+---------+------------+----------------+---------+
| test    | t1         |                |     100 |
+---------+------------+----------------+---------+
1 row in set (0.0005 sec)
```

### 3. What did you see instead (Required)

```
sql> CREATE TABLE t1(id INT PRIMARY KEY);
Query OK, 0 rows affected (0.1738 sec)

sql> SHOW STATS_HEALTHY LIKE 'test%';
+---------+------------+----------------+---------+
| Db_name | Table_name | Partition_name | Healthy |
+---------+------------+----------------+---------+
| test    | t1         |                |     100 |
+---------+------------+----------------+---------+
1 row in set (0.0005 sec)

sql> SHOW STATS_HEALTHY LIKE 'TEST%';
Empty set (0.0019 sec)
```

### 4. What is your TiDB version? (Required)

```
sql> SHOW GLOBAL VARIABLES LIKE 'lower\_case\_%';
+------------------------+-------+
| Variable_name          | Value |
+------------------------+-------+
| lower_case_file_system | 1     |
| lower_case_table_names | 2     |
+------------------------+-------+
2 rows in set (0.0023 sec)

sql> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.5.0
Edition: Community
Git Commit Hash: 706c3fa3c526cdba5b3e9f066b1a568fb96c56e3
Git Branch: heads/refs/tags/v6.5.0
UTC Build Time: 2022-12-27 03:50:44
GoVersion: go1.19.3
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv
1 row in set (0.0006 sec)
```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #41544

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by dveeden — APPROVED** [review]

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -3260,6 +3260,12 @@ func (b *PlanBuilder) buildShow(ctx context.Context, show *ast.ShowStmt) (Plan,
 			err = ErrDBaccessDenied.GenWithStackByArgs(user.AuthUsername, user.AuthHostname, mysql.SystemDB)
 		}
 		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SelectPriv, mysql.SystemDB, "", "", err)
+		if show.Tp == ast.ShowStatsHealthy {
```
Comment (tangenta) [inline planner/core/planbuilder.go:3263]: How about the other "SHOW" statements?

```
ast.ShowStatsExtended, ast.ShowStatsTopN, ast.ShowHistogramsInFlight, ast.ShowColumnStatsUsage
```

Comment (hawkingrei) [reply planner/core/planbuilder.go:3263]: We need validation test cases to fix them, so fix one first.

**Review by tangenta — APPROVED** [review]
MySQL doesn't have the syntax like `SHOW STATS_HEALTHY`. Please rename the title to ``make show `stats_healthy` case-insensitive``

**Pull request merged**